support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/ticket/registry/CassandraTicketRegistry.java

```diff
@@ -0,0 +1,254 @@
+        this.ticketCatalog = ticketCatalog;
+        this.serializer = serializer;
+        this.typeToWriteToCassandra = typeToWriteToCassandra;
+        final Cluster cluster = Cluster.builder().addContactPoints(contactPoints.split(",")).withCredentials(username, password)
+                .withProtocolVersion(ProtocolVersion.V3).build();
+
+        this.session = cluster.connect();
+
+        this.selectTgtStmt = session.prepare("select ticket from " + tgtTable + " where id = ?").setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.insertTgtStmt = session.prepare("insert into " + tgtTable + " (id, ticket, ticket_granting_ticket_id, expiration_bucket) values (?, ?, ?, ?) ")
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.deleteTgtStmt = session.prepare("delete from " + tgtTable + " where id = ?").setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.updateTgtStmt = session.prepare("update " + tgtTable + " set ticket = ? where id = ? ").setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+
+        this.selectStStmt = session.prepare("select ticket from " + stTable + " where id = ?").setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+        this.insertStStmt = session.prepare("insert into " + stTable + " (id, ticket) values (?, ?) ").setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+        this.deleteStStmt = session.prepare("delete from " + stTable + " where id = ?").setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+        this.updateStStmt = session.prepare("update " + stTable + " set ticket = ? where id = ? ").setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+
+        this.selectExStmt = session.prepare("select ticket, id from " + expiryTable + " where expiration_bucket = ? ")
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.selectDateExStmt = session.prepare("select expiration_bucket from " + expiryTable)
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+
+        this.selectLrStmt = session.prepare("select last_run from " + lastRunTable + " where id = 'LASTRUN' ")
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+        this.updateLrStmt = session.prepare("update " + lastRunTable + " set last_run = ? where id = 'LASTRUN' ")
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
+
+        final long lastRun = getLastRunTimestamp();
+        final long currentTime = currentTimeBucket();
+        if (lastRun == 0 || lastRun > currentTime) {
+            updateLastRunTimestamp(currentTime);
+        }
+    }
+
+    @Override
+    public void addTicket(final Ticket ticket) {
+        final String ticketId = ticket.getId();
+        LOGGER.debug("Inserting ticket {}", ticketId);
+        final TicketDefinition ticketDefinition = ticketCatalog.find(ticketId);
+        final String storageName = ticketDefinition.getProperties().getStorageName();
+
+        if (TGT_TABLE.equals(storageName)) {
```

Comment:
This defeats the entire purpose of the ticket catalog. Explicit checks are and should never be required.